src/v/cloud_storage/tests/util.cc

```diff
@@ -824,6 +825,89 @@ scan_result scan_remote_partition(
     };
 }
 
+std::vector<model::record_batch_header>
+scan_remote_partition_incrementally_with_closest_lso(
+  cloud_storage_fixture& imposter,
+  model::offset base,
+  model::offset max,
+  size_t maybe_max_segments,
+  size_t maybe_max_readers) {
+    ss::lowres_clock::update();
+    auto conf = imposter.get_configuration();
+    static auto bucket = cloud_storage_clients::bucket_name("bucket");
+    if (maybe_max_segments) {
+        config::shard_local_cfg()
+          .cloud_storage_max_materialized_segments_per_shard.set_value(
+            maybe_max_segments);
+    }
+    if (maybe_max_readers) {
+        config::shard_local_cfg()
+          .cloud_storage_max_segment_readers_per_shard.set_value(
+            maybe_max_readers);
+    }
```

Comment:
maybe they should be the scoped_config, to autoreset at the end of the function

Reply:
yes, but it doesn't matter that much for fuzz tests
but agree, we should do this eventually